packages/block-editor/src/components/block-list/index.js

```diff
@@ -64,6 +77,8 @@ function Root( { className, children } ) {
 				}
 			) }
 		>
+			{ element &&
+				createPortal( <BlockHeadSlot />, element.ownerDocument.head ) }
```

Comment:
Ideally it would be good if the portal can be created by the component adding the styles, but there's currently no good way to access the block ref (and the owner document). A future refactor should enable that.